jwst/ami/ami_analyze.py

```diff
@@ -41,6 +42,8 @@ def apply_LG_plus(input_model, filter_model, oversample, rotation,
         Fringe analysis data
 
     """
+    # Create copy of input_model to avoid overwriting input
+    input_copy = datamodels.ImageModel(input_model.copy())
```

Comment:
You don't need to copy it and then pass in as an `ImageModel`.  That's extra overhead that will then create a new shallow copy pointer to the new deep copy.  A simple deep copy will do I think:

```python
input_copy = input_model.copy()
```

And then the `datamodels` import is also not needed.

Reply:
Could I also remove the copy()s in the if block, as I'm now modifying the copy rather than the original?

Reply:
yes, you can then just slice `input_copy`

Reply:
Yes!  Absolutely.  I missed those.